parsec/backend/asgi/rpc.py

```diff
@@ -268,3 +281,138 @@ async def authenticated_api(raw_organization_id: str) -> Response:
+                                print("~~~~~event", event)
+                            except trio.EndOfChannel:
+                                break
+                            body_line = b"data:" + b64encode(event.dump()) + b"\n\n"
+                            await self._body_line_sender.send(body_line)
+
+                        if scope.cancelled_caught:
+                            body_line = b":keepalive\n\n"
+                            await self._body_line_sender.send(body_line)
+
+            finally:
+                # Closing sender end of the channel will cause Quart stop iterating
+                # on the receiver end and hence terminate the request
+                self._body_line_sender.close()
+
+        assert client_ctx.cancel_scope is not None
+        with client_ctx.cancel_scope:
+            async with trio.open_nursery() as nursery:
+
+                with backend.event_bus.connection_context() as client_ctx.event_bus_ctx:
+                    await backend.events.connect_events(client_ctx)
+
+                    nursery.start_soon(_events_into_body_lines)
+
+                    yield
+
+    async def __aenter__(self) -> AsyncIterable[bytes]:
+        await self._contextmanager.__aenter__()
+        return self
+
+    async def __aexit__(self, *args: Any, **kwargs: Any) -> Any:
+        return await self._contextmanager.__aexit__(*args, **kwargs)
+
+    def __aiter__(self) -> AsyncIterator[bytes]:
+        return self._body_line_receiver
+
+
+@rpc_bp.route("/authenticated/<raw_organization_id>/events", methods=["GET"])
+async def authenticated_events_api(raw_organization_id: str) -> Response:
+    backend: BackendApp = g.backend
+
+    api_version, organization_id, _, user, device = await _do_handshake(
+        raw_organization_id=raw_organization_id,
+        backend=backend,
+        allow_missing_organization=False,
+        check_authentication=True,
+        # We don't care of Content-Type given the request has no body
+        expected_content_type=None,
+        expected_accept_type=ACCEPT_TYPE_SSE,
+    )
+    assert isinstance(user, User)
+    assert isinstance(device, Device)
+
+    client_ctx = AuthenticatedClientContext(
+        api_version=api_version,
+        organization_id=organization_id,
+        device_id=device.device_id,
+        human_handle=user.human_handle,
+        device_label=device.device_label,
+        profile=user.profile,
+        public_key=user.public_key,
+        verify_key=device.verify_key,
+    )
+
+    response = current_app.response_class(
+        response=SSEResponseIterableBody(client_ctx, backend),
+        headers={
+            "Content-Type": ACCEPT_TYPE_SSE,
+            "Cache-Control": "no-cache",
+            "Transfer-Encoding": "chunked",
+        },
+        status=200,
+    )
+    # SSE are long lasting query, so we must disable Quart DOS protection here
+    response.timeout = None
```

Comment:
So timeout is not an argument `current_app.response_class()`?

Reply:
It's a hack in Quart:

```python
class Response(SansIOResponse):
    automatically_set_content_length = True
    default_mimetype = "text/html"
    data_body_class = DataBody
    file_body_class = FileBody
    implicit_sequence_conversion = True
    io_body_class = IOBody
    iterable_body_class = IterableBody
    json_module = json

    def __init__(
        self,
        response: Union[ResponseBody, AnyStr, Iterable, None] = None,
        status: Optional[int] = None,
        headers: Optional[Union[dict, Headers]] = None,
        mimetype: Optional[str] = None,
        content_type: Optional[str] = None,
    ) -> None:
        super().__init__(status, headers, mimetype, content_type)
        self.timeout: Any = Ellipsis
        [...]
```